src/class/object_users.php

```diff
@@ -480,9 +482,21 @@ function prepareData ($zData = '', $sView = 'list')
             // Submissions...
             if (lovd_isAuthorized('user', $zData['id']) === false) {
                 // Not authorized to view hidden data for this user; so we're not manager and we're not viewing ourselves. Nevermind then.
-                unset($this->aColumnsViewEntry['ownes_']);
+                unset($this->aColumnsViewEntry['ownes_'], $this->aColumnsViewEntry['auth_token_'], $this->aColumnsViewEntry['auth_token_expires_']);
             } else {
-                // Either we're viewing ourselves, or we're manager or up. Like this is easy, because now we don't need to check for the data status of the data.
+                // Either we're viewing ourselves, or we're manager or up.
+
+                // Auth token links. We don't show the token by default.
+                $zData['auth_token_'] = '(<A href="#" onclick="$.get(\'ajax/auth_token.php/' . $zData['id'] . '?view\'); return false;">Show / More information</A>)';
```

Comment:
> 404, when the client has no network connection, or when there's a timeout.

I guess failing network connection is the only real thing we would deal with, and I guess a proper message is OK even though the entire LOVD would stop to work at that time :wink: 


> We could have a simple funciton (...) a loading icon (...) and a warning icon when it fails.

That's very much overkill. This is a local (also LOVD) script that returns instantly, and it doesn't use external tools. I once happened to implement a waiting icon for such a local script, and it didn't even show because the connection was so fast (remember that showing the loading icon is yet another request, so it will always be late). Also since the chance of failing is very much zero, and if it would fail, so would the rest of LOVD, so the failure icon wouldn't load either. A simple alert that errors is more than enough, I'd say...